service/src/main/java/uk/gov/hmcts/reform/fpl/service/sdo/StandardDirectionsOrderService.java

```diff
@@ -47,20 +55,67 @@ public StandardDirectionOrder buildTemporarySDO(CaseData caseData, StandardDirec
                 previousSDO.getOrderDoc()
             );
         }
-        return StandardDirectionOrder.builder()
-            .orderDoc(document)
-            .build();
+
+        StandardDirectionOrder.StandardDirectionOrderBuilder builder = StandardDirectionOrder.builder()
+            .orderDoc(document);
+
+        if (featureToggleService.isSendNoticeOfProceedingsFromSdo()) {
+            JudgeAndLegalAdvisor judgeAndLegalAdvisor = getSelectedJudge(
+                caseData.getJudgeAndLegalAdvisor(), caseData.getAllocatedJudge()
+            );
+
+            removeAllocatedJudgeProperties(judgeAndLegalAdvisor);
+
+            builder.judgeAndLegalAdvisor(judgeAndLegalAdvisor);
+        }
+
+        return builder.build();
     }
 
     public StandardDirectionOrder buildOrderFromUpload(StandardDirectionOrder currentOrder) {
         UserInfo userInfo = idamClient.getUserInfo(requestData.authorisation());
 
-        return StandardDirectionOrder.builder()
+        StandardDirectionOrder.StandardDirectionOrderBuilder builder = StandardDirectionOrder.builder()
             .orderStatus(currentOrder.getOrderStatus())
             .dateOfUpload(time.now().toLocalDate())
             .uploader(userInfo.getName())
-            .orderDoc(prepareOrderDocument(currentOrder.getOrderDoc(), currentOrder.getOrderStatus()))
-            .build();
+            .orderDoc(prepareOrderDocument(currentOrder.getOrderDoc(), currentOrder.getOrderStatus()));
+
+        if (featureToggleService.isSendNoticeOfProceedingsFromSdo()) {
+            builder.judgeAndLegalAdvisor(currentOrder.getJudgeAndLegalAdvisor());
+        }
+
+        return builder.build();
+    }
+
+    public JudgeAndLegalAdvisor getJudgeAndLegalAdvisorFromSDO(CaseData caseData) {
+        StandardDirectionOrder standardDirectionOrder = caseData.getStandardDirectionOrder();
+
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = JudgeAndLegalAdvisor.builder().build();
+
+        if (standardDirectionOrder != null && isNotEmpty(standardDirectionOrder.getJudgeAndLegalAdvisor())) {
+            judgeAndLegalAdvisor = standardDirectionOrder.getJudgeAndLegalAdvisor();
+        }
+
+        if (isNotEmpty(caseData.getAllocatedJudge())) {
+            judgeAndLegalAdvisor = prepareSDOJudgeFields(caseData);
+        }
+
+        return judgeAndLegalAdvisor;
+    }
```

Comment:
Difficult method to work with. Inherited from code base just a lift and shift. Not too keen at this current time to refactor but plan to create a refactor ticket. Existing TODO in Standard directions order controller should be actioned.